### DISCUSSION TITLE
useDocument with Route Params empty path error

### DISCUSSION THREAD

**Discussion by Darkade** [vuejs/vuefire#1416] [Questions and Help] [answered]
I'm trying to replicate this example with Vue3, Firestore and Vite. I want to use params as part of the path for a document; however the `useDocument` line is throwing an error on the browser: "Uncaught FirebaseError: Function doc() cannot be called with an empty path" but I'm not understanding why

```ts
const route = useRoute()
// since route is reactive, `contactSource` will be reactive too
const contactSource = computed(() =>
  doc(collection(db, 'contacts'), route.params.id)
)
// contact will always be in sync with the data source
const contact = useDocument(contactSource)
```
https://github.com/vuejs/vuefire/blob/6ec8a6ca6a96a917edae51a2f6a4924e46472b37/docs/guide/realtime-data.md?plain=1#L69

this is my actual code
```
const route = useRoute();
const characterSource = computed(
  ()=> {
    const game = collection(db, "game");
    const room = doc(game, route.params.room as string);
    const characters = collection(room, "characters");
    return doc(characters, route.params.player as string);
  }
);

const character = useDocument(characterSource);
```

My routes are working and I can access their values both in the template and in the setup script

Thanks!!

**Comment by posva** [comment, marked as answer]
The error means that `route.params.room` is empty at some point. You can verify it within your computed and `return null` if it's empty, VueFire will still work with `null` as the `characterSource`
